File: app/src/main/java/net/onest/time/components/holder/AdapterHolder.java
package net.onest.time.components.holder;

import android.widget.BaseExpandableListAdapter;

import androidx.recyclerview.widget.RecyclerView;

import net.onest.time.adapter.todo.TodoItemAdapter;
import net.onest.time.api.vo.TaskVo;

import java.util.List;

public class AdapterHolder {
    private BaseExpandableListAdapter baseExpandableListAdapter;
    private RecyclerView.Adapter<RecyclerView.ViewHolder> recyclerViewAdapter;
//    private TodoItemAdapter todoItemAdapter;

    public AdapterHolder(RecyclerView.Adapter<RecyclerView.ViewHolder> adapter) {
        this.recyclerViewAdapter = adapter;
    }

//    public AdapterHolder(TodoItemAdapter todoItemAdapter) {
//        this.todoItemAdapter = todoItemAdapter;
//    }

    public AdapterHolder(BaseExpandableListAdapter adapter) {
        this.baseExpandableListAdapter = adapter;
    }

    public void notifyDataSetChanged() {
        if (baseExpandableListAdapter != null) {
            baseExpandableListAdapter.notifyDataSetChanged();
            return;
        }

        if (recyclerViewAdapter != null) {
            recyclerViewAdapter.notifyDataSetChanged();
            return;
        }

    }

//    public void notifyItemChanged(int position, TaskVo taskVo) {
//        List<TaskVo> itemListByDay = todoItemAdapter.getItemListByDay();
//        itemListByDay.set(position,taskVo);
//        todoItemAdapter.setItemListByDay(itemListByDay);
//        todoItemAdapter.notifyItemChanged(position);
//    }
    public void notifyItemChanged(int position) {
        if (recyclerViewAdapter != null) {
            recyclerViewAdapter.notifyItemChanged(position);
        }
    }
}

File: app/src/main/java/net/onest/time/components/holder/AdapterHolder.kt
package net.onest.time.components.holder

import android.annotation.SuppressLint
import android.widget.BaseExpandableListAdapter
import androidx.recyclerview.widget.RecyclerView

class AdapterHolder {
    private var baseExpandableListAdapter: BaseExpandableListAdapter? = null
    private var recyclerViewAdapter: RecyclerView.Adapter<RecyclerView.ViewHolder>? = null

    constructor(adapter: RecyclerView.Adapter<RecyclerView.ViewHolder>?) {
        this.recyclerViewAdapter = adapter
    }

    constructor(adapter: BaseExpandableListAdapter?) {
        this.baseExpandableListAdapter = adapter
    }

    @SuppressLint("NotifyDataSetChanged")
    fun notifyDataSetChanged() {
        baseExpandableListAdapter?.notifyDataSetChanged()

        recyclerViewAdapter?.notifyDataSetChanged()
    }

    fun notifyItemChanged(position: Int) {
        baseExpandableListAdapter?.notifyDataSetChanged()

        recyclerViewAdapter?.notifyItemChanged(position)
    }

    fun notifyItemRemoved(position: Int) {
        baseExpandableListAdapter?.notifyDataSetChanged()

        recyclerViewAdapter?.notifyItemRemoved(position)
    }

    fun notifyItemInserted(position: Int) {
        baseExpandableListAdapter?.notifyDataSetChanged()

        recyclerViewAdapter?.notifyItemInserted(position)
    }
}
